lib/pushmi_pullyu/aip_downloader.rb

```diff
@@ -0,0 +1,147 @@
+require 'pushmi_pullyu/aip_downloader_files_directories'
+require 'pushmi_pullyu/aip_logger'
+require 'pushmi_pullyu/fedora_object_fetcher'
+require 'pushmi_pullyu/solr_fetcher'
+
+# Download all of the metadata/datastreams and associated data
+# related to an object
+
+class PushmiPullyu::AipDownloader
+
+  include PushmiPullyu::AipDownloaderFilesDirectories
+
+  attr_reader :noid, :config, :logger, :aip_logger, :fetcher,
+              :aipcreation_log
+
+  attr_accessor :got_characterization, :got_thumbnail,
+                :got_fedora3foxml, :got_fedora3foxml_metadata
+
+  def initialize(noid, config = nil, application_logger = nil)
+    @noid = noid
+
+    @config = config || PushmiPullyu.options
+    @logger = application_logger || PushmiPullyu.logger
+    @fetcher = PushmiPullyu::FedoraObjectFetcher.new(self.noid, self.config)
+
+    initialize_directory_names
+    initialize_filenames
+
+    # We need to log to application log and to the AIP log
+    @aip_logger = PushmiPullyu::AipLogger.new(@noid, aipcreation_log, @logger)
```

Comment:
We might be arguing at cross-purposes here. When I say "I don't know what logger it's writing to" in another comment, I mean, it's unclear because we're tossing different loggers around through these instance variables and method parameters, sometimes shadowing the global logger through lazily initiated cache variables, so it's not clear at that call site what logger is an instance of, and the fact that it's a method param implies that it's unstable between calls.

What I'm saying would be clearer would be a situation in which all logging goes through the singleton `PusmiPullyu.logger`, since that was the entire motivation for creating the singleton in the first place. If you call `PusmiPullyu.logger.log_preservation_action('blah blah blah')` here, while it's true that I may not know which file(s) that call writes to, I would at least be clear on where to look without inspecting this method parameter at runtime to try to sort out what's happen where.